src/main/java/com/netflix/blitz4j/LoggingConfiguration.java

```diff
@@ -266,9 +283,9 @@ public void stop() {
      * @see com.netflix.config.PropertyListener#addProperty(java.lang.Object,
      * java.lang.String, java.lang.Object, boolean)
      */
-    public void addProperty(Object source, String name, Object value, boolean beforeUpdate) {
-        if (shouldProcessProperty(name, beforeUpdate)) {
-            updatedProps.put(name, value);
+    public synchronized void addProperty(Object source, String name, Object value, boolean beforeUpdate) {
+        if (beforeUpdate == false && isLog4JProperty(name)) {
```

Comment:
should beforeUpdate == false happen inside a function here since it's used in a couple places this way?